WordPress/gutenberg
# Gallery block: remove the v1 code and add deprecations to auto transform the v1 galleries

## The gallery experimental flag is going to be removed in 11.9, and this change will be merged for 12.0 and cherry picked into the 5.9 release. Due to the breaking nature of the gallery change we want to stage the rollout and have one release of the plugin with the new Gallery block enabled by default for new blocks, but no migration of old content.

## Description

Removes the gallery v1 code and adds deprecations to auto upgrade v1 galleries on edit

Fixes: #32075

## Testing

- Before checking out the PR, on an existing local dev env disable the gallery experiment and add a number of v1 galleries of different sizes, column numbers, linkTo and image size options
- Check out this PR and enable the gallery experiment
- Load the previously created galleries in the editor and make sure they are automatically migrated to the new format correctly
- Make sure that `npm run text-unit` and `npm run fixtures:regenerate` complete successfully
- Test manual transformations to and from Jetpack gallery and slideshow blocks, and from gallery shortcodes (easy way to get valid shortcodes is to add a gallery in the classic block)
